packages/extension-ui/src/components/Dropdown.tsx

```diff
@@ -68,16 +81,48 @@ function Dropdown({
             )
           )}
         </select>
+        <FontAwesomeIcon
+          className={`icon ${isLocked ? 'icon-locked' : ''}`}
+          icon={faChevronDown}
+        />
+        <div
+          className='locked-container'
+          onClick={_toggleLocked}
+        >
+          {isLocked ? (
+            <Svg
+              className='locked'
+              src={lockedIcon}
+            />
+          ) : (
+            <Svg
+              className='locked'
+              src={unlockedIcon}
+            />
+          )}
+        </div>
```

Comment:
Maybe it would make sense to make a small component out of this (even in the same file) that emits true/false onClick and then here you could just use setLocked directly. I feel like this functionality could be encapsulated outside and it would make flow a bit prettier (without those nasty callbacks :))
(also the className `locked` for both SVGs is a bit semantically strange - maybe just `locked-icon` or similar?